tests/backup/backup_psa_test.go

```diff
@@ -548,3 +551,606 @@ var _ = Describe("{EnableNsAndClusterLevelPSAWithBackupAndRestore}", Label(TestC
+			baselineNamespaceOnDest := fmt.Sprintf("single-ns-multi-app-%v-baseline-label", RandomString(3))
+			nsPSALabelMap[baselineNamespaceOnDest] = BaselinePSALabel
+
+			for ns, psaLabel := range nsPSALabelMap {
+				err = CreateNamespaceAndAssignLabels(ns, psaLabel)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Creating namespace %v and assigning  %v labels", ns, psaLabel))
+			}
+
+			//  switch cluster context to source cluster
+			log.InfoD("Switching cluster context back to source cluster")
+			err = SetSourceKubeConfig()
+			log.FailOnError(err, "Failed to set source kubeconfig")
+
+			namespaceMapping := make(map[string]string)
+			namespaceMapping[restrictedNamespaceInSrcCluster] = baselineNamespaceOnDest
+			namespaceMapping[baselineNamespaceInSrcCluster] = privilegedNamespaceOnDestOne
+			namespaceMapping[privilegedNamespaceInSrcCluster] = privilegedNamespaceOnDestTwo
+			log.Infof("Namespace mapping is %v", namespaceMapping)
+
+			//  Taking restore of the backup taken
+			log.InfoD("Restoring backup of multiple namespace with different PSA set with namespace mapping")
+			customRestoreWithNamespaceAndStorageClassMapping = fmt.Sprintf("%s-%v-ns-mapping-%v", RestoreNamePrefix, RandomString(3), backupNameWithMultipleNsHavingDiffPSALabel)
+			err = CreateRestoreWithValidation(ctx, customRestoreWithNamespaceAndStorageClassMapping, backupNameWithMultipleNsHavingDiffPSALabel, namespaceMapping, nil, DestinationClusterName, BackupOrgID, allScheduledAppContexts)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying %s backup's restore %s creation on multiple NS with different PSA label of higher privilege with namespace mapping %v and storage class mapping %v", backupNameWithMultipleNsHavingDiffPSALabel, customRestoreWithNamespaceAndStorageClassMapping, nil, nil))
+			restoreList = append(restoreList, customRestoreWithNamespaceAndStorageClassMapping)
+		})
+
+		Step("Create backup of multiple NS with different PSA set and restore in default way", func() {
+			//  Taking restore of the backup taken
+			log.InfoD("Restoring backup of multiple namespaces with different PSA set in default")
+			customRestoreWithNamespaceAndStorageClassMapping = fmt.Sprintf("%s-%v-ns-mapping-%v", RestoreNamePrefix, RandomString(3), backupNameWithMultipleNsHavingDiffPSALabel)
+			err = CreateRestoreWithValidation(ctx, customRestoreWithNamespaceAndStorageClassMapping, backupNameWithMultipleNsHavingDiffPSALabel, nil, nil, SourceClusterName, BackupOrgID, allScheduledAppContexts)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying %s backup's restore %s creation on multiple NS with defferent PAS label with namespace mapping %v and storage class mapping %v", backupNameWithMultipleNsHavingDiffPSALabel, customRestoreWithNamespaceAndStorageClassMapping, nil, nil))
+			restoreList = append(restoreList, customRestoreWithNamespaceAndStorageClassMapping)
+		})
+	})
+
+	JustAfterEach(func() {
+		defer EndPxBackupTorpedoTest(scheduledAppContexts)
+		defer func() {
```

Comment:
can you please move this to justbefore each , please refer RestoreFromHigherPrivilegedNamespaceToLower test 

cc : @mkoppal-px

Reply:
@sn-px in this test RestoreFromHigherPrivilegedNamespaceToLower the applications are being deployed in JustBeforeEach stage. But In my case I am deploying the applications based on the requirement on the scenario to be tested like whenever I want take backup of multiple namespaces of restricted type. That is the reason I added it in AfterEach

Reply:
ok